### PR TITLE
fix: incorrect peerDependency on "constructs"

### PR DESCRIPTION
**Pull request by RomainMuller**
Both `@monocdk-experiment/assert` and `aws-cdk-lib` had an outdated peer
dependency declaration against `construct` (at `^3.0.4` when the code
actually depends on features introduced as recently as `^3.2.0`). This
corrects the declaration accordingly.

Additionally, fixes the pack prerequisite check for .NET to accept
.NET 5.0 as a valid provider for .NET Core 3.1 (because it is).

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@monocdk-experiment/assert/package.json [hunk 1]
```diff
@@ -47,7 +47,7 @@
     "@aws-cdk/cloudformation-diff": "0.0.0"
   },
   "peerDependencies": {
-    "constructs": "^3.0.4",
+    "constructs": "^3.2.0",
```
Comment (nija-at) [inline packages/@monocdk-experiment/assert/package.json:50]: Should we have a linter check too? Or is this managed by dependabot and was missed?

Comment (nija-at) [reply packages/@monocdk-experiment/assert/package.json:50]: Looks like we explicitly don't auto-upgrade constructs - https://github.com/aws/aws-cdk/blob/4553d4b37cc7c2062eebc36bfe7a03b0226b598d/.github/workflows/yarn-upgrade.yml#L48

Linter check probably so we don't screw up when we upgrade again.

Comment (nija-at) [reply packages/@monocdk-experiment/assert/package.json:50]: Note about adding the linter check. We use a different version of constructs for v2, so do take that into account.

You'll need to use a different version depending on the MV. something like - https://github.com/aws/aws-cdk/blob/4553d4b37cc7c2062eebc36bfe7a03b0226b598d/tools/pkglint/lib/rules.ts#L1531

File: tools/pkglint/lib/rules.ts [hunk 2]
```diff
@@ -617,6 +617,38 @@ export class NoPeerDependenciesMonocdk extends ValidationRule {
...
 }
 
+/**
+ * Validates that the same version of `constructs` is used wherever a dependency
+ * is specified, so that they must all be udpated at the same time (through an
+ * update to this rule).
+ *
+ * Note: v1 and v2 use different versions respectively.
+ */
+export class ConstructsConsistentDependencyVersion extends ValidationRule {
```
Comment (nija-at) [inline tools/pkglint/lib/rules.ts:627]: Oh my, such long names. How about just `ConstructsVersion`?

File: tools/pkglint/lib/rules.ts [hunk 3]
```diff
@@ -617,6 +617,38 @@ export class NoPeerDependenciesMonocdk extends ValidationRule {
...
+  public readonly name = 'peer-deps/constructs';
```
Comment (nija-at) [inline tools/pkglint/lib/rules.ts:628]: since you're checking all dependencies

```suggestion
  public readonly name = 'deps/constructs';
```

**Review by nija-at — APPROVED** [review]
Provisional approval. 

Make sure the discussion about breaking change is concluded on our Slack channel.

File: packages/@monocdk-experiment/assert/package.json [hunk 4]
```diff
@@ -47,7 +47,7 @@
     "@aws-cdk/cloudformation-diff": "0.0.0"
   },
   "peerDependencies": {
-    "constructs": "^3.0.4",
```
Comment (eladb) [inline packages/@monocdk-experiment/assert/package.json:50]: We normally recommend to declare a pinned devDependency in order to ensure that the code gets tested with the minimal version required by the peer. Is this not the case here? Why?

Copying @nija-at and @NetaNir whom
I talked to about this.

File: tools/pkglint/lib/rules.ts (hunk 2 above)
Comment (RomainMuller) [reply tools/pkglint/lib/rules.ts:627]:
```suggestion
export class ConstructsVersion extends ValidationRule {
```

**Pull request merged**